state-sync/storage-service/types/src/lib.rs

```diff
@@ -343,9 +494,25 @@ impl<T: Copy + Ord> CompleteDataRange<T> {
         self.highest
     }
 
-    /// Returns true iff the given data item is within this range
-    pub fn contains(&self, data_item: T) -> bool {
-        data_item >= self.lowest && data_item <= self.highest
+    /// Returns the length of the data range.
+    ///
+    /// Note that `CompleteDataRange` maintains the invariant that the range
```

Comment:
Future nit: Use checked arithmetic inside the function (even if it's impossible for overflows). That way we avoid the long comment here and the debug assert.